### DISCUSSION TITLE
Private NuGet package issues with "dotnet lambda publish" and AOT

### DISCUSSION THREAD

**Discussion by dguisinger** [aws/aws-lambda-dotnet#1712] [Q&A] [answered]
I am stuck trying to transition from the .NET 8.0 preview release to .NET 8.0 final release.

As part of the transition, we decided to switch to AOT.  We are using CodeBuild with Amazon Linux 2023 to build (AMAZON_LINUX_2_5).  My understanding from Issue 1161 is that if you are not using Amazon Linux, it uses Docker to do the build.  I would prefer not to have lambda publish "magically" running everything inside a docker container I don't control, so I am confused as to why this didn't switch to using the native host for building.

The overall problem I am trying to solve is I use CodeArtifact for several private NuGet packages.  I don't control the AOT build container, its completely managed by "dotnet lambda publish", so I have absolutely no idea how to get CodeArtifact credentials from the root level CodeBuild container into the lambda publish container.

I haven't been able to find any documentation around this "feature".

The only reference I have found to this was someone suggesting creating a docker file....but I already had a dockerfile from when I was building on with the pre-release versions of .NET 8 and its clearly not using the docker file.

**Comment by ashishdhingra** [comment, marked as answer]
I guess this has been addressed as part of https://github.com/aws/aws-lambda-dotnet/issues/1718. Marking this discussion as answered.
